crypto/jwx.go

```diff
@@ -323,7 +323,7 @@ func EncryptJWE(payload []byte, protectedHeaders map[string]interface{}, publicK
 		enc = headers.ContentEncryption()
 	}
 
-	encoded, err := jwe.Encrypt(payload, alg, publicKey, enc, headers.Compression(), jwe.WithProtectedHeaders(headers))
+	encoded, err := jwe.Encrypt(payload, jwe.WithKey(alg, publicKey), jwe.WithContentEncryption(enc), jwe.WithCompress(jwa.Deflate), jwe.WithProtectedHeaders(headers))
```

Comment:
Currently, compression is derived from the unmarshalled headers. And if not set, no compression is applied. This changes that to always "Deflate"? (change of behavior)